utest/test_argument_validation.py

```diff
@@ -0,0 +1,166 @@
+import io
+import unittest
+import pathlib
+from io import StringIO
+from unittest.mock import patch
+from robocop.config import Config
+from robocop.version import __version__
+
+
+class TestArgumentValidation(unittest.TestCase):
+    def setUp(self):
+        self.config = Config()
+
+    def test_prog_name(self):
+        self.assertEqual(self.config.parser.prog, 'robocop')
+
+    def test_parser_default_help_disabled(self):
+        self.assertFalse(self.config.parser.add_help)
+
+    def test_default_args(self):
+        self.assertSetEqual(self.config.filetypes, {'.resource', '.robot'})
+        self.assertSetEqual(self.config.include, set())
+        self.assertSetEqual(self.config.exclude, set())
+        self.assertSetEqual(self.config.reports, set())
+        self.assertListEqual(self.config.configure, [])
+        self.assertEqual(self.config.format, "{source}:{line}:{col} [{severity}] {msg_id} {desc}")
+        self.assertListEqual(self.config.paths, [])
+        self.assertIsNone(self.config.output)
+
+    def test_default_args_after_parse(self):
+        args = self.config.parse_opts([''])
+        self.assertSetEqual(args.filetypes, {'.resource', '.robot'})
+        self.assertSetEqual(args.include, set())
+        self.assertSetEqual(args.exclude, set())
+        self.assertSetEqual(args.reports, set())
+        self.assertListEqual(args.configure, [])
+        self.assertEqual(args.format, "{source}:{line}:{col} [{severity}] {msg_id} {desc}")
+        self.assertListEqual(args.paths, [''])
+        self.assertIsNone(args.output)
+
+    def test_overwrite_default_filetypes(self):
+        args = self.config.parse_opts(['--filetypes', 'txt', ''])
+        self.assertSetEqual(args.filetypes, {'.txt'})
+
+    def test_duplicate_default_filetypes(self):
```

Comment:
we could use name of argument in the beggining so it's easy to see what feature is tested. like we have "test_include.." in this case "test_filetypes_duplicate_default", "test_filetypes_overwrite_default" and so on. what do you think?

Or maybe there is some way to tag test cases

Reply:
I think it's a good idea but I will look into unittest a little bit more. 
We can also use pytest instead of unittest but need to find out if that is not an overkill for us.